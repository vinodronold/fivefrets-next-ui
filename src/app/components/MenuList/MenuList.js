import React, { PureComponent } from 'react'
import { bool } from 'prop-types'
import { color } from '../..//constants/theme'
import Button from '../Button'

export default class MenuList extends PureComponent {
  static propTypes = {
    isMenuOpen: bool.isRequired
  }
  render() {
    const { isMenuOpen } = this.props
    return (
      <div>
        <Button>Home</Button>
        <Button>Browse</Button>
        <Button>Login</Button>
        <style jsx>{`
          div {
            display: flex;
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            margin: auto;
            overflow: hidden;
            justify-content: center;
            text-align: center;
            flex-direction: column;
            color: ${color.primary()};
            background-color: ${color.bg()};
            transition: all 0.5s;
          }
        `}</style>
        <style jsx>{`
          div {
            opacity: ${isMenuOpen ? 1 : 0};
            z-index: ${isMenuOpen ? 10 : -1};
          }
        `}</style>
      </div>
    )
  }
}
