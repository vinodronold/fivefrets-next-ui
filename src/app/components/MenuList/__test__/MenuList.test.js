import React from 'react'
import renderer from 'react-test-renderer'
import MenuList, { onMenuClickHandler, onRouteChangeCompleteHandler } from '../MenuList'

jest.mock('next/router')

describe('components/MenuList', () => {
  it('should match snapshot when Menu is closed', () => {
    const _closed_menu_list = renderer.create(<MenuList isMenuOpen={false} ToggleMenu={jest.fn()} />)
    expect(_closed_menu_list).toMatchSnapshot()
  })
  it('should match snapshot when Menu is open', () => {
    const _open_menu_list = renderer.create(<MenuList isMenuOpen={true} ToggleMenu={jest.fn()} />)
    expect(_open_menu_list).toMatchSnapshot()
  })

  it('routes and ToggleMenu should be called in onMenuClickHandler', () => {
    const _ToggleMenu = jest.fn()
    const _test_route = {
      href: jest.fn(),
      as: jest.fn()
    }
    onMenuClickHandler(_ToggleMenu, _test_route)()
    expect(_test_route.href).toBeCalled()
    expect(_test_route.as).toBeCalled()
    onRouteChangeCompleteHandler(_ToggleMenu)()
    expect(_ToggleMenu).toBeCalled()
  })
})
