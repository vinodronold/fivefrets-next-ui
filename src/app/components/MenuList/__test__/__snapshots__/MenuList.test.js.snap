// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MenuList should match snapshot when Menu is closed 1`] = `
<div
  className="jsx-2435928121 jsx-2330443265"
>
  <button
    className="jsx-67710159 jsx-4255320455 "
    disabled={false}
    onClick={[Function]}
  >
    Home
  </button>
  <button
    className="jsx-67710159 jsx-4255320455 "
    disabled={false}
    onClick={[Function]}
  >
    Browse
  </button>
  <button
    className="jsx-67710159 jsx-4255320455 "
    disabled={false}
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`components/MenuList should match snapshot when Menu is open 1`] = `
<div
  className="jsx-2435928121 jsx-410418109"
>
  <button
    className="jsx-67710159 jsx-4255320455 "
    disabled={false}
    onClick={[Function]}
  >
    Home
  </button>
  <button
    className="jsx-67710159 jsx-4255320455 "
    disabled={false}
    onClick={[Function]}
  >
    Browse
  </button>
  <button
    className="jsx-67710159 jsx-4255320455 "
    disabled={false}
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
